psqlextra/partitioning/plan.py

```diff
@@ -37,14 +38,26 @@ def apply(self, using: Optional[str]) -> None:
 
         connection = connections[using or "default"]
 
-        with transaction.atomic():
-            with connection.schema_editor() as schema_editor:
-                for partition in self.creations:
-                    partition.create(
-                        self.config.model,
-                        schema_editor,
-                        comment=AUTO_PARTITIONED_COMMENT,
-                    )
+        if self.creations:
+            with transaction.atomic():
+                with connection.schema_editor() as schema_editor:
+                    for partition in self.creations:
+                        partition.create(
+                            self.config.model,
+                            schema_editor,
+                            comment=AUTO_PARTITIONED_COMMENT,
+                        )
+
+        if self.deferred_creations:
+            with transaction.atomic():
+                with connection.schema_editor() as schema_editor:
+                    for partition in self.creations:
```

Comment:
Shouldn't this iterate over `self.deferred_creations`

Reply:
Yes, great catch, thanks!